### PR TITLE
Reduce de complexity of the js-ipfs-api module

### PR DESCRIPTION
**Pull request by diasdavid**
> **WIP**

Right now, the guts of js-ipfs-api are overly complex, making it hard to debug, to apply changes or even to develop, since we have to run every single test every single time. To make the experience more pleasant, this PR has the goals to:

- [x] Break the tests into individual units (not requiring to run them all at once)
- [x] Make the request API receive an object, instead of several arguments (https://github.com/ipfs/js-ipfs-api/pull/307#discussion_r69274864)

What other things would you like to see that would make using the js-ipfs-code more sane?

### CODE REVIEW

File: src/request-api.js [hunk 1]
```diff
@@ -84,7 +85,7 @@ function requestAPI (config, path, args, qs, files, buffer, cb) {
   const port = config.port ? `:${config.port}` : ''
 
   const opts = {
-    method: files ? 'POST' : 'GET',
+    method: 'POST',
```
Comment (dignifiedquire) [inline src/request-api.js]: Not all requests should be `POST` requests

Comment (diasdavid) [inline src/request-api.js]: Let's check with @whyrusleeping, I've changed this after a conversation with him when we were debugging some npm-on-ipfs stuff.

Comment (diasdavid) [inline src/request-api.js]: ping @whyrusleeping

Comment (diasdavid) [inline src/request-api.js]: 🛎

**Comment by diasdavid** [comment]
@VictorBjelkholm moving away from the argument flags :)  #43a1dfa

**Comment by diasdavid** [comment]
I've refactored all the test units that not require multiple nodes, to be come isolated. Will handle the remaining (3) with the interface-ipfs-core tests.

**Comment by dignifiedquire** [comment]
Why u no wait for CR?

**Comment by diasdavid** [comment]
It was not merged to master, it was merged to the other PR branch ;)

**Comment by dignifiedquire** [comment]
ahh okay :)

**Pull request merged**